lib/debug/debugger_tooltip.dart

```diff
@@ -0,0 +1,200 @@
+library atom.debugger_tooltip;
+
+import 'dart:async';
+import 'dart:math' as math;
+
+import 'package:atom/node/workspace.dart';
+import 'package:atom/utils/disposable.dart';
+import 'package:logging/logging.dart';
+import 'package:petitparser/petitparser.dart';
+
+import '../analysis_server.dart';
+import '../elements.dart';
+import '../impl/tooltip.dart';
+import '../material.dart';
+import '../state.dart';
+import '../utils.dart';
+
+import './debugger_ui.dart';
+import './model.dart';
+
+final Logger _logger = new Logger('atom.debugger_tooltip');
+
+class DebugTooltipManager implements Disposable {
+
+  final Property<int> endOffset = new Property();
+
+  Parser reverseContextParser;
+
+
+  DebugTooltipManager() {
+    var expression = undefined();
```

Comment:
Is `undefined()` from `package:petitparser`? Perhaps, use a type annotation here instead of `var`, or import petitparser with a prefix to make that clear?

Reply:
Done.